packtools/sps/formats/oai_dc_agris.py

```diff
@@ -0,0 +1,689 @@
+                        <dc:title xml:lang="es">
+                            <![CDATA[ La canción reflexiva: en torno al estatuto crítico de la música popular en Brasil ]]>
+                        </dc:title>
+                    </ags:resource>
+                </ags:resources>
+            </metadata>
+        </record>
+    """
+    add_title(xml_oai_dc_agris, xml_tree)
+
+
+def xml_oai_dc_agris_creator_pipe(xml_oai_dc_agris, xml_tree):
+    """
+    Schema (https://agris.fao.org/agris_ods/dlio.dtd.txt):
+        <!-- ELEMENT creator -->
+        <!ELEMENT dc:creator (#PCDATA | ags:creatorPersonal | ags:creatorCorporate | ags:creatorConference)*>
+        <!ELEMENT ags:creatorPersonal (#PCDATA)>
+        <!ELEMENT ags:creatorCorporate (#PCDATA)>
+        <!ELEMENT ags:creatorConference (#PCDATA)>
+
+    Example:
+        <record>
+            <metadata>
+                <ags:resources>
+                    <ags:resource ags:ARN="XS2021000111">
+                        <dc:creator>
+                            <ags:creatorPersonal>de-Oliveira-Gerolamo, Ismael</ags:creatorPersonal>
+                        </dc:creator>
+                    </ags:resource>
+                </ags:resources>
+            </metadata>
+        </record>
+    """
+    author = article_authors.Authors(xml_tree)
+    try:
+        surname = author.contribs[0].get('surname')
+        given_name = author.contribs[0].get('given_names')
+        author_name = f' {surname.strip()}, {given_name.strip()} '
+        add_creator(xml_oai_dc_agris, author_name.strip())
+    except IndexError:
+        pass
+
+
+def add_publisher(xml_oai_dc_agris, publisher):
+    try:
+        ags = ET.Element('{http://purl.org/agmes/1.1/}publisherName')
+        ags.text = publisher.publishers_names[0].strip()
+
+        dc = ET.Element('{http://purl.org/dc/elements/1.1/}publisher')
+        dc.append(ags)
+
+        xml_oai_dc_agris.find(".//ags:resource", {"ags": "http://purl.org/agmes/1.1/"}).append(dc)
+    except IndexError:
+        pass
+
+
+def xml_oai_dc_agris_publisher_pipe(xml_oai_dc_agris, xml_tree):
+    """
+    Schema (https://agris.fao.org/agris_ods/dlio.dtd.txt):
+        <!-- ELEMENT publisher -->
+        <!ELEMENT dc:publisher (ags:publisherName | ags:publisherPlace)*>
+        <!ELEMENT ags:publisherName (#PCDATA)>
+        <!ELEMENT ags:publisherPlace (#PCDATA)>
+
+    Example:
+        <dc:publisher>
+            <ags:publisherName>Pontificia Universidad Católica de Chile, Facultad de Filosofía, Instituto de Estética</ags:publisherName>
+        </dc:publisher>
+    """
+    publisher = journal_meta.Publisher(xml_tree)
+
+    add_publisher(xml_oai_dc_agris, publisher)
+
+
+def get_date(dt, m=False, d=False):
+    try:
+        year = dt.pub_dates[0].get('year')
+        month = dt.pub_dates[0].get('month')
+        day = dt.pub_dates[0].get('day')
+        exceptions = [None, '', '0', '00']
+
+        if year is None:
+            return
```

Comment:
Estas linhas podem ficar na linha 293, de modo que se year is None, você economiza no processamento das linhas 294 a 296

Reply:
@robertatakenaka corrigido.